addons/rating/models/rating_mixin.py

```diff
@@ -132,20 +132,23 @@ def _rating_get_repartition(self, add_stats=False, domain=None):
         base_domain = expression.AND([self._rating_domain(), [('rating', '>=', 1)]])
         if domain:
             base_domain += domain
-        rg_data = self.env['rating.rating']._read_group(base_domain, ['rating'], ['__count'])
+        rg_data = self.env['rating.rating']._read_group(base_domain, ['rating', 'res_id'], ['__count'])
         # init dict with all possible rate value, except 0 (no value for the rating)
-        values = dict.fromkeys(range(1, 6), 0)
-        for rating, count in rg_data:
+        values = {rec.id: dict.fromkeys(range(1, 6), 0) for rec in self}
+        for rating, rec_id, count in rg_data:
```

Comment:
should be `res_id` because it's easily confusing with the id of the rating otherwise. Point is that you probably confused them yourself.